sql/handler.cc

```diff
@@ -7314,6 +7307,12 @@ bool Vers_parse_info::fix_alter_info(THD *thd, Alter_info *alter_info,
     }
   }
 
+  if (share->tmp_table && share->tmp_table != INTERNAL_TMP_TABLE)
+  {
+    my_error(ER_VERS_TEMPORARY, MYF(0), "ADD SYSTEM VERSIONING");
```

Comment:
It is not about "ADD SYSTEM VERSIONING". We prohibit all versioning operations on user temporary tables.